src/hooks/useShortcut.js

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useCallback, useReducer } from 'react'
+
+const keysReducer = (state, action) => {
+  switch (action.type) {
+    case 'SET_KEY_DOWN':
+      const keyDownState = { ...state, [action.key]: true }
+      return keyDownState
+    case 'SET_KEY_UP':
+      const keyUpState = { ...state, [action.key]: false }
+      return keyUpState
+    case 'RESET_KEY':
+      const resetState = { ...action.data }
+      return resetState
+    default:
+      return state
+  }
+}
+
+const useShortcut = (shortcutKeys, callback) => {
+  if (!Array.isArray(shortcutKeys))
+    throw new Error(
+      'The first parameter to `useKeyboardShortcut` must be an ordered array of `KeyboardEvent.key` strings.'
+    )
+
+  if (!shortcutKeys.length)
+    throw new Error(
+      'The first parameter to `useKeyboardShortcut` must contain atleast one `KeyboardEvent.key` string.'
+    )
+
+  if (typeof callback !== 'function')
+    throw new Error(
```

Comment:
I like that you consider all this cases. For me it would be fine without this errors as it's unlikely that you make those mistakes in your own code considering the few places where this is used. But keep it, if you like.